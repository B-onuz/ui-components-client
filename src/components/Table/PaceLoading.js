import React from 'react'
import styled from 'styled-components'
import Spinner from '../Spinner'

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(255, 255, 255, 0.6);
  z-index: 999;
  & > div {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
  }
`

export default () => {
  return (
    <Wrapper>
      <Spinner color={'primary'} />
    </Wrapper>
  )
}
