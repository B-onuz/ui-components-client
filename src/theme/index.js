const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
