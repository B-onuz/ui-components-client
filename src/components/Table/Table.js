import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { space, layout, compose } from 'styled-system'
import PaceLoading from './PaceLoading'
import shortid from 'shortid'

const StyledTable = styled.table`
  border-spacing: 0;
  position: relative;
  border-collapse: collapse;
  width: 100%;
`

const TableHead = styled.thead`
  background: #f1f3f8;
  display: table-header-group;
`

const TableHeader = styled.th`
  text-transform: uppercase;
  font-size: 0.8em;
  color: #838383;
  font-weight: 400;
  padding: 1em;
  position: relative;
  position: sticky;
  z-index: 2;
  top: 0;
  background: #f1f3f8;
  ${({ width }) =>
    !!width &&
    `
    width: ${width};
  `}
  ${({ minWidth }) =>
    minWidth &&
    `
    min-width: ${minWidth}px;
  `}
  /* padding: 13px 40px; */
  padding: 10px 20px;
  ${({ order }) =>
    order &&
    `
    cursor: pointer;
  `}
  ${({ align }) =>
    align &&
    `
    text-align: ${align};
  `}
`

const OrderArrow = styled.span`
  position: relative;
  width: 20px;
  height: 20px;
  margin-left: 4px;
  @media only screen and (max-width: 500px) {
    transform: rotateZ(90deg);
  }
  ${({ isCurrent }) =>
    !!isCurrent
      ? `
  &::before,
  &::after {
    content: '';
    display: block;
    position: absolute;
    width: 1px;
    height: 8px;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
    background-color: #000;
    transition: transform 0.25s ease-in-out 0s;
  }`
      : `
    &::before,
    &::after {
      content: '';
      display: block;
      border: solid #838383;
      border-width: 0 1px 1px 0;
      display: inline-block;
      padding: 3px;
      position: absolute;
      left: 0;
      right: 0;
      margin: auto;
      width: 0px;
      height: 0px;
    }
    &::after {
      top: 4px;
      transform: rotate(-135deg);
    }

    &::before {
      bottom: 4px;
      transform: rotate(45deg);
    }
  `}

  ${({ order, isCurrent }) => {
    if (!!isCurrent) {
      return order === 'asc'
        ? `
      &::before {
        transform: translate3d(-2px, 0px, 0px) rotate(45deg);
      }
      &::after {
        transform: translate3d(3px, 0px, 0px) rotate(-45deg);
      }
    `
        : `
      &::before {
        transform: translate3d(3px, 0px, 0px) rotate(45deg);
      }
      &::after {
        transform: translate3d(-2px, 0px, 0px) rotate(-45deg);
      }
    `
    } else {
      return ''
    }
  }}
`

const TableRow = styled.tr`
  /* padding: 13px 40px; */
  @media only screen and (max-width: 500px) {
    min-width: 150px;
    font-size: 14px;
  }
`

const TableHeaderSpan = styled.span`
  display: inline-flex;
  align-items: center;
  flex-direction: inherit;
  justify-content: flex-start;
  ${({ width }) =>
    !!width &&
    `
    width: ${width};
  `}
  ${({ minWidth }) =>
    minWidth &&
    `
    min-width: ${minWidth}px;
  `}
`

const TableData = styled.td`
  ${({ align }) =>
    align &&
    `
    text-align: ${align};
  `}
  padding: 10px 20px;
  @media only screen and (max-width: 500px) {
    padding: 12px 8px;
    text-align: center;
  }
`

const WrapperTable = styled.div`
  overflow-x: auto;
  position: relative;
  height: 100%;
  width: 100%;
  flex: 1;
`

const Wrapper = styled.div`
  position: relative;
  border: 1px solid #efefef;
  display: flex;
  flex-direction: column;
  flex: ${({ flex }) => flex};
  ${compose(space, layout)}
`

const TableBody = styled.tbody`
  position: relative;
  display: table-row-group;
  & > ${TableRow} {
    background-color: #f8fafc;
    &:nth-child(odd) {
      background-color: #fff;
    }
  }
`

const Table = ({ children, headers, pagination, onChangeOrder, data, flex, loading, ...rest }) => {
  const handleChangeOrder = (item) => {
    if (onChangeOrder) {
      onChangeOrder(item)
    }
  }

  const getDataByKey = ({ item, key }) => {
    const keys = key.split('.')
    return keys.reduce((acc, item) => {
      return acc[item] || ''
    }, item)
  }

  return (
    <Wrapper flex={flex} {...rest}>
      {!!loading && <PaceLoading />}
      <WrapperTable loading={loading}>
        <StyledTable>
          <TableHead>
            <TableRow>
              {(headers || []).map((item) => (
                <TableHeader
                  width={item.width}
                  minWidth={item.minWidth}
                  key={`${shortid.generate()}`}
                  order={item.sort}
                  align={item.align}
                  onClick={() => item.sort && handleChangeOrder(item)}
                >
                  <TableHeaderSpan>
                    {item.title} {item.sort && <OrderArrow isCurrent={!!(pagination.sort === item.key)} order={pagination.order} />}
                  </TableHeaderSpan>
                </TableHeader>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {!(data || []).length && (
              <TableRow>
                <TableData style={{ textAlign: 'center' }} colSpan={(headers || []).length}>
                  {!!loading ? 'Carregando...' : 'Nenhum registro encontrado'}
                </TableData>
              </TableRow>
            )}
            {(data || []).map((row) => (
              <TableRow key={`${shortid.generate()}`}>
                {headers.map((header) =>
                  header.cellComponent ? (
                    <TableData width={header.width} align={header.align} key={`${shortid.generate()}`}>
                      <header.cellComponent row={row} header={header} value={getDataByKey({ key: header.key, item: row })} />
                    </TableData>
                  ) : (
                    <TableData align={header.align} key={`${shortid.generate()}`}>
                      {getDataByKey({ key: header.key, item: row })}
                    </TableData>
                  )
                )}
              </TableRow>
            ))}
          </TableBody>
        </StyledTable>
      </WrapperTable>
    </Wrapper>
  )
}

Table.defaultProps = {}

Table.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.object),
}

export default Table
