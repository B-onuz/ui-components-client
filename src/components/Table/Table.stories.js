import React, { useState } from 'react'
import Table from './Table'
import Button from '../Button'
import Input from '../Input'
import ContextMenu from '../ContextMenu'
import { action } from '@storybook/addon-actions'
import { faPlus, faCloudUploadAlt } from '@fortawesome/free-solid-svg-icons'

export default {
  title: 'Table',
  component: Table,
}

const functionA = action('Função A')

const functionB = action('Função B')

const functionC = action('Função C')

const contextOptions = [
  {
    text: 'editar',
    buttonFunction: functionA,
  },
  {
    text: 'adicionar',
    buttonFunction: functionB,
  },
  {
    text: 'excluir',
    buttonFunction: functionC,
  },
]

export const TableExample = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Table
      width={[1]}
      onChangeOrder={handleChangeOrder}
      pagination={{
        sort,
        order,
      }}
      headers={[
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Especialidades',
          key: 'especialidades',
          sort: true,
          align: 'left',
          width: '350px',
        },
        {
          title: 'Tarefas',
          key: 'tarefas',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
          width: '50px',
        },
      ]}
      data={[
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Last Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
      ]}
    />
  )
}

export const TableExampleMaxHeight = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Table
      width={[2 / 3]}
      onChangeOrder={handleChangeOrder}
      maxHeight={200}
      pagination={{
        sort,
        order,
      }}
      headers={[
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Especialidades',
          key: 'especialidades',
          sort: true,
          align: 'left',
        },
        {
          title: 'Tarefas',
          key: 'tarefas',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ]}
      data={[
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Last Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
      ]}
    />
  )
}

export const TableLoadingWithData = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Table
      width={[2 / 3]}
      onChangeOrder={handleChangeOrder}
      maxHeight={200}
      loading={true}
      pagination={{
        sort,
        order,
      }}
      headers={[
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Especialidades',
          key: 'especialidades',
          sort: true,
          align: 'left',
        },
        {
          title: 'Tarefas',
          key: 'tarefas',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ]}
      data={[
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        { nome: 'Last Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
      ]}
    />
  )
}
export const TableLoadingDynamicWithData = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')
  const [loading, setLoading] = useState(false)
  const [loadingTime, setLoadingTime] = useState(300)

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  const handleLoadingButton = () => {
    setLoading(true)
    setTimeout(() => {
      setLoading(false)
    }, 5000)
  }

  const handleChangeLoadingTime = (e) => {
    setLoadingTime(e.target.value)
  }

  return (
    <>
      <Input value={loadingTime} onChange={handleChangeLoadingTime} />
      <Button color={'primary'} onClick={handleLoadingButton}>
        Carregar
      </Button>
      <Table
        width={[2 / 3]}
        onChangeOrder={handleChangeOrder}
        maxHeight={200}
        loading={loading}
        pagination={{
          sort,
          order,
        }}
        headers={[
          {
            title: 'Advogados',
            key: 'nome',
            sort: true,
            align: 'left',
          },
          {
            title: 'Especialidades',
            key: 'especialidades',
            sort: true,
            align: 'left',
          },
          {
            title: 'Tarefas',
            key: 'tarefas',
            sort: true,
            align: 'center',
          },
          {
            title: 'Ações',
            key: 'actions',
            align: 'center',
          },
        ]}
        data={[
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { nome: 'Last Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        ]}
      />
    </>
  )
}

export const TableWithCellComponent = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <>
      <Table
        width={[2 / 3]}
        onChangeOrder={handleChangeOrder}
        // maxHeight={200}
        pagination={{
          sort,
          order,
        }}
        headers={[
          {
            title: 'Advogados',
            key: 'nome',
            sort: true,
            align: 'left',
          },
          {
            title: 'Especialidades',
            key: 'especialidades',
            sort: true,
            align: 'left',
            minWidth: 500,
          },
          {
            title: 'Tarefas',
            key: 'tarefas',
            sort: true,
            cellComponent({ row, headers, value }) {
              return (
                <span style={{ color: 'red', textDecoration: 'underline' }}>
                  {value} - {row._id}
                </span>
              )
            },
            align: 'center',
          },
          {
            title: 'Ações',
            key: 'actions',
            align: 'center',
            cellComponent({ row, headers, value }) {
              return <ContextMenu contextMenuActions={contextOptions} contextFunctions buttonIcon={faPlus} direction={'right'} />
            },
          },
        ]}
        data={[
          { _id: '1', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '2', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '3', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '4', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '5', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '6', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '7', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '8', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '9', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '10', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '11', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '12', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '13', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '14', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '15', nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          { _id: '16', nome: 'Last Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
        ]}
      />
    </>
  )
}

export const TableEmpty = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Table
      width={[2 / 3]}
      onChangeOrder={handleChangeOrder}
      maxHeight={200}
      pagination={{
        sort,
        order,
      }}
      headers={[
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Especialidades',
          key: 'especialidades',
          sort: true,
          align: 'left',
        },
        {
          title: 'Tarefas',
          key: 'tarefas',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ]}
      data={[]}
    />
  )
}
export const LoadingWithEmpty = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Table
      width={[2 / 3]}
      onChangeOrder={handleChangeOrder}
      maxHeight={200}
      loading={true}
      pagination={{
        sort,
        order,
      }}
      headers={[
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Especialidades',
          key: 'especialidades',
          sort: true,
          align: 'left',
        },
        {
          title: 'Tarefas',
          key: 'tarefas',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ]}
      data={[]}
    />
  )
}
