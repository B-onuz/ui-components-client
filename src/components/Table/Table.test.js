import React, { useState } from 'react'
import Table from './Table.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Table changes the class when hovered', () => {
  const TableExample = () => {
    const [sort, setSort] = useState('nome')
    const [order, setOrder] = useState('asc')

    const handleChangeOrder = (item) => {
      if (item.key === sort) {
        setOrder(order === 'asc' ? 'desc' : 'asc')
      } else {
        setSort(item.key)
        setOrder('asc')
      }
    }

    return (
      <ThemeProvider theme={theme}>
        <Table
          width={[2 / 3]}
          onChangeOrder={handleChangeOrder}
          pagination={{
            sort,
            order,
          }}
          headers={[
            {
              title: 'Advogados',
              key: 'nome',
              sort: true,
              align: 'left',
            },
            {
              title: 'Especialidades',
              key: 'especialidades',
              sort: true,
              align: 'left',
            },
            {
              title: 'Tarefas',
              key: 'tarefas',
              sort: true,
              align: 'center',
            },
            {
              title: 'Ações',
              key: 'actions',
              align: 'center',
            },
          ]}
          data={[
            { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
            { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
            { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
            { nome: 'Nome do advogado', especialidades: 'Comercial, Contratos', tarefas: 10 },
          ]}
        />
      </ThemeProvider>
    )
  }
  const component = renderer.create(<TableExample />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
