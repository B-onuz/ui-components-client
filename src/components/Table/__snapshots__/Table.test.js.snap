// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table changes the class when hovered 1`] = `
.c2 {
  border-spacing: 0;
  position: relative;
  border-collapse: collapse;
  width: 100%;
}

.c3 {
  background: #f1f3f8;
  display: table-header-group;
}

.c6 {
  text-transform: uppercase;
  font-size: 0.8em;
  color: #838383;
  font-weight: 400;
  padding: 1em;
  position: relative;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  top: 0;
  background: #f1f3f8;
  padding: 13px 40px;
  cursor: pointer;
  text-align: left;
}

.c10 {
  text-transform: uppercase;
  font-size: 0.8em;
  color: #838383;
  font-weight: 400;
  padding: 1em;
  position: relative;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  top: 0;
  background: #f1f3f8;
  padding: 13px 40px;
  cursor: pointer;
  text-align: center;
}

.c11 {
  text-transform: uppercase;
  font-size: 0.8em;
  color: #838383;
  font-weight: 400;
  padding: 1em;
  position: relative;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  top: 0;
  background: #f1f3f8;
  padding: 13px 40px;
  text-align: center;
}

.c8 {
  position: relative;
  width: 20px;
  height: 20px;
  margin-left: 4px;
}

.c8::before,
.c8::after {
  content: '';
  display: block;
  position: absolute;
  width: 1px;
  height: 8px;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  background-color: #000;
  -webkit-transition: -webkit-transform 0.25s ease-in-out 0s;
  -webkit-transition: transform 0.25s ease-in-out 0s;
  transition: transform 0.25s ease-in-out 0s;
}

.c8::before {
  -webkit-transform: translate3d(-2px,0px,0px) rotate(45deg);
  -ms-transform: translate3d(-2px,0px,0px) rotate(45deg);
  transform: translate3d(-2px,0px,0px) rotate(45deg);
}

.c8::after {
  -webkit-transform: translate3d(3px,0px,0px) rotate(-45deg);
  -ms-transform: translate3d(3px,0px,0px) rotate(-45deg);
  transform: translate3d(3px,0px,0px) rotate(-45deg);
}

.c9 {
  position: relative;
  width: 20px;
  height: 20px;
  margin-left: 4px;
}

.c9::before,
.c9::after {
  content: '';
  display: block;
  border: solid #838383;
  border-width: 0 1px 1px 0;
  display: inline-block;
  padding: 3px;
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
  width: 0px;
  height: 0px;
}

.c9::after {
  top: 4px;
  -webkit-transform: rotate(-135deg);
  -ms-transform: rotate(-135deg);
  transform: rotate(-135deg);
}

.c9::before {
  bottom: 4px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: inherit;
  -ms-flex-direction: inherit;
  flex-direction: inherit;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c13 {
  text-align: left;
  padding: 13px 40px;
}

.c14 {
  text-align: center;
  padding: 13px 40px;
}

.c1 {
  overflow-x: auto;
  position: relative;
  height: 100%;
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  position: relative;
  border: 1px solid #efefef;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 66.66666666666666%;
}

.c12 {
  position: relative;
  display: table-row-group;
}

.c12 > .c4 {
  background-color: #f8fafc;
}

.c12 > .c4:nth-child(odd) {
  background-color: #fff;
}

@media only screen and (max-width:500px) {
  .c8 {
    -webkit-transform: rotateZ(90deg);
    -ms-transform: rotateZ(90deg);
    transform: rotateZ(90deg);
  }
}

@media only screen and (max-width:500px) {
  .c9 {
    -webkit-transform: rotateZ(90deg);
    -ms-transform: rotateZ(90deg);
    transform: rotateZ(90deg);
  }
}

@media only screen and (max-width:500px) {
  .c5 {
    min-width: 150px;
    font-size: 14px;
  }
}

@media only screen and (max-width:500px) {
  .c13 {
    padding: 12px 8px;
    text-align: center;
  }
}

@media only screen and (max-width:500px) {
  .c14 {
    padding: 12px 8px;
    text-align: center;
  }
}

<div
  className="c0"
  width={
    Array [
      0.6666666666666666,
    ]
  }
>
  <div
    className="c1"
  >
    <table
      className="c2"
    >
      <thead
        className="c3"
      >
        <tr
          className="c4 c5"
        >
          <th
            className="c6"
            onClick={[Function]}
            order={true}
          >
            <span
              className="c7"
            >
              Advogados
               
              <span
                className="c8"
                order="asc"
              />
            </span>
          </th>
          <th
            className="c6"
            onClick={[Function]}
            order={true}
          >
            <span
              className="c7"
            >
              Especialidades
               
              <span
                className="c9"
                order="asc"
              />
            </span>
          </th>
          <th
            className="c10"
            onClick={[Function]}
            order={true}
          >
            <span
              className="c7"
            >
              Tarefas
               
              <span
                className="c9"
                order="asc"
              />
            </span>
          </th>
          <th
            className="c11"
            onClick={[Function]}
          >
            <span
              className="c7"
            >
              Ações
               
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        className="c12"
      >
        <tr
          className="c4 c5"
        >
          <td
            className="c13"
          >
            Nome do advogado
          </td>
          <td
            className="c13"
          >
            Comercial, Contratos
          </td>
          <td
            className="c14"
          >
            10
          </td>
          <td
            className="c14"
          >
            
          </td>
        </tr>
        <tr
          className="c4 c5"
        >
          <td
            className="c13"
          >
            Nome do advogado
          </td>
          <td
            className="c13"
          >
            Comercial, Contratos
          </td>
          <td
            className="c14"
          >
            10
          </td>
          <td
            className="c14"
          >
            
          </td>
        </tr>
        <tr
          className="c4 c5"
        >
          <td
            className="c13"
          >
            Nome do advogado
          </td>
          <td
            className="c13"
          >
            Comercial, Contratos
          </td>
          <td
            className="c14"
          >
            10
          </td>
          <td
            className="c14"
          >
            
          </td>
        </tr>
        <tr
          className="c4 c5"
        >
          <td
            className="c13"
          >
            Nome do advogado
          </td>
          <td
            className="c13"
          >
            Comercial, Contratos
          </td>
          <td
            className="c14"
          >
            10
          </td>
          <td
            className="c14"
          >
            
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
